spacemesh/v2/network.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package spacemesh.v2;
+
+message NetworkInfoRequest {}
+
+message NetworkInfoResponse {
+  uint64 genesis_time = 1; // network genesis time as unix epoch time
+  uint32 epoch_num_layers = 2; // number of layers per epoch
+  uint32 layer_duration = 3; // layer duration, in seconds
+  bytes genesis_id = 4;
+  string hrp = 5;
+  uint32 first_genesis = 7; // first effective genesis layer
```

Comment:
I see in the code that this has to do with checkpoint restore. In practice effective genesis and "first genesis" are the same. Let's keep things simple and collapse these two into a single data item called `effective_genesis_layer` for clarity.